Mathlib/Algebra/Lie/Weights/Killing.lean

```diff
@@ -351,6 +396,86 @@ lemma isCompl_ker_weight_span_coroot (α : Weight K H L) :
     replace hα : corootSpace α ≠ ⊥ := by simpa using hα
     rwa [ne_eq, ← LieSubmodule.coe_toSubmodule_eq_iff] at hα
 
+lemma exists_isSl2Triple_of_weight_isNonZero {α : Weight K H L} (hα : α.IsNonZero) :
+    ∃ h e f : L, IsSl2Triple h e f ∧ e ∈ rootSpace H α ∧ f ∈ rootSpace H (- α) := by
+  obtain ⟨e, heα : e ∈ rootSpace H α, he₀ : e ≠ 0⟩ := α.exists_ne_zero
+  obtain ⟨f, hfα, hf⟩ : ∃ f ∈ rootSpace H (-α), killingForm K L e f ≠ 0 := by
+    contrapose! he₀
+    simpa using mem_ker_killingForm_of_mem_rootSpace_of_forall_rootSpace_neg K L H heα he₀
+  have hef := lie_eq_killingForm_smul_of_mem_rootSpace_of_mem_rootSpace_neg heα hfα
+  let h : H := ⟨⁅e, f⁆, hef ▸ Submodule.smul_mem _ _ (Submodule.coe_mem _)⟩
+  have hh : α h ≠ 0 := by
+    change (α : H →ₗ[K] K) h ≠ 0
+    have : h = killingForm K L e f • (cartanEquivDual H).symm α := by simp [Subtype.ext_iff, hef]
+    rw [this, map_smul, smul_eq_mul, ne_eq, mul_eq_zero, not_or]
+    exact ⟨hf, root_apply_cartanEquivDual_symm_ne_zero hα⟩
+  let f' := (2 * (α h)⁻¹) • f
```

Comment:
What do you think of swapping the names of `f` and `f'` to indicate which one is tmp and which on is the "final candidate"?